light-client/src/tests.rs

```diff
@@ -129,16 +208,46 @@ impl From<Provider<AnonLightBlock>> for Provider<LightBlock> {
     }
 }
 
+impl From<WitnessProvider<AnonLightBlock>> for WitnessProvider<LightBlock> {
+    fn from(p: WitnessProvider<AnonLightBlock>) -> Self {
+        Self {
+            value: p.value.into(),
+        }
+    }
+}
+
+pub fn random_peer_id(count: usize) -> PeerId {
+    let peer_ids: Vec<PeerId> = vec![
+        "ADBEEFC0FFEEFACADEBADFADADE0BEFEEDC0C0AD".parse().unwrap(),
+        "BADFADBEEFC0FFEEFACADEAD0BEFEEDC0C0ADEAD".parse().unwrap(),
+        "CADEEDC0C0ADEADBEEFBADFADAD0BEFEC0FFEEFA".parse().unwrap(),
+        "D0BEFEEDC0C0ADEABADFADADBEEFC0FFEEFACADE".parse().unwrap(),
+        "EEFC0FFEEFACADEBADFADAD0BEFEEDC0C0ADEADB".parse().unwrap(),
+        "FC0FFEEFABAC0C0ADEADDFBEEADAD0BEFEEDCADE".parse().unwrap(),
+    ];
+    peer_ids[count]
+}
+
 impl From<TestBisection<AnonLightBlock>> for TestBisection<LightBlock> {
     fn from(tb: TestBisection<AnonLightBlock>) -> Self {
+        let mut witnesses: Vec<WitnessProvider<LightBlock>> = tb.witnesses.into_iter().map(Into::into).collect();
+
+        for (count, provider) in witnesses.iter_mut().enumerate() {
+            for lb in provider.value.lite_blocks.iter_mut() {
+                lb.provider = random_peer_id(count);
```

Comment:
Is this random or is it deterministic? 🤔

Reply:
This is actually deterministic: https://github.com/informalsystems/tendermint-rs/pull/371/files#diff-11a3f1d34a7a9515595b37dba6035f46R222-R232

Reply:
Initially, it was to be used for something else and then I didn't think about renaming it. Thanks for the catch @brapse !